### PR TITLE
updates: thealtf4stream begins work on Docker neovim env

### PR DESCRIPTION
**Pull request by erkrnt**
WORK IN PROGRESS!

The goal here is to essentially have a Neovim in Docker environment that I can use anywhere that has Docker. This will be a drop in replacement for the local `neovim` binary I install on any of my development hosts as well.

### CODE REVIEW

File: TheAltF4Stream/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  automake \
+  autoconf \
+  libtool \
+  pkgconf \
+  coreutils \
+  curl \
+  unzip \
+  gettext-tiny-dev \
+  git
+RUN git clone --branch "nightly" --progress "https://github.com/neovim/neovim.git" \
```
Comment (asbjornhaland) [inline TheAltF4Stream/Dockerfile:14]: You should use '--dept=1' for the git clone to speed up the clone time. You don't care about the whole history.

Comment (erkrnt) [reply TheAltF4Stream/Dockerfile:14]: Appreciate it ! Will update once I get back!

**Comment by ThePrimeagen** [comment]
this is pretty cool idea.

**Comment by bwisdom88** [comment]
I think I have a solution if you would be interested in seeing it. The one thing I dont have solved is tree-sitter has to load everytime. Im thinking for that you will have to restrict which parsers are loaded, but since that is a decision youll have to make I left that one alone.

**Comment by erkrnt** [comment]
@bwisdom88 so I think using Docker to mount the `.config` via a volume would allow you to cache it so you only would technically have to run it on changes.

**Comment by erkrnt** [comment]
This is as close as I'm going to get it for now - wouldn't mind others taking a look at the implementation and improving it. I'm going to be testing all three setups locally and opening new issues as I discover things.

I also left some questionable implementations in the code to see if others have opinions about them.

**Pull request merged**